src/Compilers/Core/MSBuildTaskTests/TargetTests.cs

```diff
@@ -0,0 +1,479 @@
+
+            var customMeta = item.Metadata.SingleOrDefault(m => m.Name == metaName.EvaluatedValue);
+            Assert.NotNull(customMeta);
+            Assert.Equal("abc", customMeta.EvaluatedValue);
+        }
+
+        [Fact]
+        public void GenerateAnalyzerConfigCoreHandlesMissingMetadata()
+        {
+            XmlReader xmlReader = XmlReader.Create(new StringReader($@"
+<Project>
+    <Import Project=""Microsoft.Managed.Core.targets"" />
+    <ItemGroup>
+        <Compile Include=""file1.cs"" />
+    </ItemGroup>
+    <ItemGroup>
+        <AnalyzerItemMetadata Include=""Compile"" MetadataName=""CustomMeta"" />
+        <AnalyzerItemMetadata Include=""Compile2"" MetadataName=""CustomMeta"" />
+    </ItemGroup>
+</Project>
+"));
+
+            var instance = CreateProjectInstance(xmlReader);
+
+            bool runSuccess = instance.Build(target: "GenerateMSBuildAnalyzerConfigFile", GetTestLoggers());
+            Assert.True(runSuccess);
+
+            var items = instance.GetItems("_GeneratedAnalyzerConfigMetadata");
+            Assert.Single(items);
+
+            var item = items.Single();
+            Assert.Equal("_GeneratedAnalyzerConfigMetadata", item.ItemType);
+
+            var itemType = item.Metadata.SingleOrDefault(m => m.Name == "ItemType");
+            Assert.NotNull(itemType);
+            Assert.Equal("Compile", itemType.EvaluatedValue);
+
+            var metaName = item.Metadata.SingleOrDefault(m => m.Name == "MetadataName");
+            Assert.NotNull(metaName);
+            Assert.Equal("CustomMeta", metaName.EvaluatedValue);
+        }
+
+        [Fact]
+        public void GenerateAnalyzerConfigCoreHandlesMalformedAnalyzerItemMetadata()
+        {
+            XmlReader xmlReader = XmlReader.Create(new StringReader($@"
+<Project>
+    <Import Project=""Microsoft.Managed.Core.targets"" />
+    <ItemGroup>
+        <Compile Include=""file1.cs"" />
+    </ItemGroup>
+    <ItemGroup>
+        <AnalyzerItemMetadata Include=""Compile"" />
+    </ItemGroup>
+</Project>
+"));
+
+            var instance = CreateProjectInstance(xmlReader);
+
+            bool runSuccess = instance.Build(target: "GenerateMSBuildAnalyzerConfigFile", GetTestLoggers());
+            Assert.True(runSuccess);
+
+            var items = instance.GetItems("_GeneratedAnalyzerConfigMetadata");
+            Assert.Single(items);
+
+            var item = items.Single();
+            Assert.Equal("_GeneratedAnalyzerConfigMetadata", item.ItemType);
+
+            var itemType = item.Metadata.SingleOrDefault(m => m.Name == "ItemType");
+            Assert.NotNull(itemType);
+            Assert.Equal("Compile", itemType.EvaluatedValue);
+
+            var metaName = item.Metadata.SingleOrDefault(m => m.Name == "MetadataName");
+            Assert.NotNull(metaName);
+            Assert.Equal("", metaName.EvaluatedValue);
+        }
+
+        private ProjectInstance CreateProjectInstance(XmlReader reader)
```

Comment:
How does this work with respect to loading the common tasks and targets? Essentially what targets get loaded using this approach and from where do they get loaded?

Reply:
It doesn't load anything by default (including even the normal MSBuild ones). The tests import or shim the things they need / want to test 

---
In reply to: [426954574](https://github.com/dotnet/roslyn/pull/44365#discussion_r426954574) [](ancestors = 426954574)